packages/flame/lib/src/sprite_batch.dart

```diff
@@ -419,27 +432,38 @@ class SpriteBatch {
       return;
     }
 
-    final renderPaint = paint ?? _emptyPaint;
-
-    final hasNoColors = _colors.every((c) => c == _defaultColor);
-    final actualBlendMode = blendMode ?? defaultBlendMode;
-    if (!hasNoColors && actualBlendMode == null) {
-      throw 'When setting any colors, a blend mode must be provided.';
-    }
+    paint ??= _emptyPaint;
 
     if (useAtlas && !_flippedAtlasStatus.isGenerating) {
+      final transforms = _batchItems.values
+          .map((e) => e.transform)
+          .toList(growable: false);
+      final sources = _batchItems.values
+          .map((e) => e.source)
+          .toList(growable: false);
+      final colors = _batchItems.values
+          .map((e) => e.color)
```

Comment:
Along with my previous [comment](https://github.com/flame-engine/flame/pull/3650/files#r2417214005), if you apply the flip transformation on the sources here, it should work. But I don't think it is a good idea to do this in the render method.

```suggestion
          .map((e) => e.flip
              ? Rect.fromLTWH(
                  // The atlas is twice as wide when the flipped atlas is generated.
                  (atlas.width * (_flippedAtlasStatus.isGenerated ? 1 : 2)) -
                      e.source.right,
                  e.source.top,
                  e.source.width,
                  e.source.height,
                )
              : e.source)
```